src/core/options-registry.cpp

```diff
@@ -0,0 +1,122 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+#include "options-registry.h"
+#include "basics.h"
+#include <src/librealsense-exception.h>
+#include <vector>
+#include <mutex>
+#include <map>
+
+
+namespace librealsense {
+
+
+LRS_EXTENSION_API std::string const & get_string( rs2_option );
+
+
+namespace by_name {
+
+
+static std::map< std::string, rs2_option > _option_by_name = []()
+{
+    // Automatically add all our known options into the option names -- that way, no duplicate custom option
+    // can be added...
+    std::map< std::string, rs2_option > map;
+    for( auto option = rs2_option( 0 ); option < RS2_OPTION_COUNT; option = rs2_option( option + 1 ) )
+        map[get_string( option )] = option;
+    return std::move( map );
+}();
+static std::vector< std::string > _name_by_index;
+static std::mutex _mutex;
+
+
+inline size_t option_to_index( rs2_option by_name_option )
+{
+    return -by_name_option - 1;
+}
+
+
+inline rs2_option index_to_option( size_t index )
+{
+    return rs2_option( -( int( index ) + 1 ) );
```

Comment:
Consider checking if index >= INT_MAX.

Reply:
Same as the other

Reply:
I must say this function is not readable.
I asked Remi to look and he also didn't understand.
Why all the values are negative?

@maloel please add some comments

Reply:
Added:
```
    // Index = index into the _name_by_index array -> 0-based
    // Registered rs2_option value will be negative, and starting at -1:
```
Will be in next PR